src/components/App.tsx

```diff
@@ -14,26 +13,64 @@ import Map from './map/Map';
 import Sidebar from './sidebar/Sidebar';
 import NodeWindow from './NodeWindow';
 import * as s from './app.scss';
-const rootPath: string = '/';
+import { Route, RouteComponentProps } from 'react-router-dom';
+import { LineStore } from '../stores/lineStore';
+import LineService from '../services/lineService';
+import RouteService from '../services/routeService';
+import { RouteStore } from '../stores/routeStore';
+import * as qs from 'qs';
 
 interface IAppState {
     showLogin: boolean;
 }
 
-interface IAppProps {
+interface IAppProps extends RouteComponentProps<any> {
     notificationStore?: NotificationStore;
     loginStore?: LoginStore;
     sidebarStore?: SidebarStore;
     mapStore?: MapStore;
+    lineStore?: LineStore;
+    routeStore?: RouteStore;
 }
 
-@inject('mapStore')
-@inject('notificationStore')
-@inject('sidebarStore')
-@inject('loginStore')
+@inject('mapStore', 'notificationStore', 'sidebarStore', 'loginStore', 'lineStore', 'routeStore')
 @observer
 class App extends React.Component<IAppProps, IAppState> {
 
+    componentDidMount() {
+        this.queryAll();
+    }
+    componentWillReceiveProps(props: any) {
+        this.queryAll();
+    }
+
+    async queryAll() {
+        this.props.sidebarStore!.isLoading = true;
+        await this.queryAllLines();
+        await this.queryRoutes();
+        this.props.sidebarStore!.isLoading = false;
+    }
```

Comment:
Doesn't this mean that we are querying all lines and routes, every time the url changes? I know that Apollo doesn't actually send the request do to caching, but still? Overall I don't think that the App component should take care of fetching things that some of it children might need.

Reply:
I do somewhat agree.. Let's discuss this at the office